apis/mobilite.py

```diff
@@ -0,0 +1,134 @@
+import json
+import requests
+import logging
+import pandas as pd
+
+from datetime import datetime, timedelta
+
+from db.records import get_record_dict
+from utils.errors import FreskError
+from utils.keywords import *
+from utils.language import detect_language_code
+from utils.location import get_address
+
+
+def get_json_from_source(source):
+    try:
+        response = requests.get(source)
+        # Check if the request was successful (status code 200)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            logging.info(f"Request failed with status code: {response.status_code}")
+    except requests.RequestException as e:
+        logging.info(f"An error occurred: {e}")
+
+
+def get_mobilite_data(source):
+    logging.info("Getting data from Fresque de la Mobilité API")
+
+    records = []
+
+    # Workaround to get two make results and merge them (will change in the future)
+    json_sessions = get_json_from_source("https://hook.eu1.make.com/ui9bvl4c3w69dxdlb7goskl3o22x74um")
+    df_sessions = pd.json_normalize(json_sessions["response"]["results"])
+    json_versions = get_json_from_source("https://hook.eu1.make.com/sy4ud6vxutts9h62t4tt6gv0xr5rrkyd")
+    df_versions = pd.json_normalize(json_versions["response"]["results"])
+    df_sessions = df_sessions.merge(df_versions, how='left', left_on="atelier_version_custom_atelier_version", right_on="_id", suffixes=(None, "_y"))
+
+    # Suppress sessions if there are cancelled
+    if 'annulation_custom_annulation' in df_sessions.columns:
+        df_sessions = df_sessions[df_sessions['annulation_custom_annulation'].isnull()]
+
+    for event_id, row in df_sessions.iterrows():
+        logging.info("")
+
+        online = is_online(row["format_option_version_format"])
+        training = is_training(row["type_option_version_type"])
+        description = ""
+        kids = is_for_kids(row["p_rim_tre_option_version_p_rim_tre"])
+        sold_out = (row["nb_places_number"] - row["nb_participants_number"]) == 0
+        source_link = tickets_link = f'https://app.fresquedelamobilite.org/atelier_details/{row["_id"]}'
+        title = f'{row["type_option_version_type"]} {row["th_me_option_version_th_me"]} {row["p_rim_tre_option_version_p_rim_tre"]} {row["format_option_version_format"]}'
```

Comment:
I'd be more comfortable if these fields were checked before accessing them. If one of them doesn't exist, please skip the current row with an error message.